docs/developer-guidelines.md

```diff
@@ -0,0 +1,195 @@
+directory](./).
+
+### Testing
+
+* [Testing document](testing.md)
+
+Testing helps to ensure the quality of the code base. Every pull request
+submitted should first be tested to ensure that all existing tests pass. If
+changes could affect state external to HaaS's DB (network switches and
+headnodes), then deployment tests should also be run.
+
+When introducing new functionality, new tests (both unit and more comprehensive)
+should be added that provide adequate coverage.
+
+If fixing a bug, a regression test should accompany the bug fix to ensure that
+the bug does not return.
+
+## Pull Requests
+
+Once the checklist above has been met, issue a [pull request][pr] from your
+personal fork to the [main HaaS repo][repo]. 
+
+Labels are the way we communicate github pull request and issue status. They should be assigned in line with the *Suggestions* section
+of [this wiki
+page](https://github.com/CCI-MOC/haas/wiki/Issue-Tracker-Proposal#suggestions).
+
+## Code review
+
+Code reviews help increase code quality by finding:
+* mistakes that can oftentimes be overlooked by a single developer
+* improving readability
+* helping reviewers to learn about different areas of the code
+
+Reviewers are the final guardians for good code quality. If you see a piece of
+complex code, that is probably where you want to spend a lot of your review
+time. Remember the 80/20 rule (80% of the bugs come from 20% of the code).
+
+## Friendliness
+
+We want the MOC HaaS to be a pleasure to work with. Thus, while we want to
+ensure a high level of code quality, we don't want new project contributors to
+get overwhelmed. This should be kept in mind while conducting a review.  In
+some cases, such as missing tests, it may help a new contributor for an
+experienced reviewer to work closely or even write the initial tests
+themselves.
+
+As a practical matter, if feedback is lengthy and would take a longer time to
+write up than simply to do, it can be considered a courtesy to check out the
+submitter's branch, make the changes yourself, then submit a Pull Request
+directly to the submitter's branch. This is especially helpful for
+documentation PRs. When doing so, it can help to notify the submitter directly
+as they may otherwise miss your pending PR (they likely won't be expecting it).
+
+## Approval criteria
+
+Pull Requests require at least 2 successful reviews, also known as "+1"'s, in
+order to be merged, with at least 2 core developers being involved in some way:
+either as a submitter or a reviewer.
+
+The exception to this is documentation, where we are a little more lenient in
+the interest of lowering the barrier to having better docs. For docs changes,
+typically one +1 is sufficient.
+
+Whomever provides the enabling +1 is responsible for clicking the merge button
+on github. If you do not have commit access, then please add your +1 to the PR
+and ask one of the Core Developers to complete the merge.
+
+## Labels
```

Comment:
I like the idea of having an assignee, however github only allows one and we typically require two reviewers.

Additionally, it's not necessarily a bad things if more than 1 or 2 reviewers review a request.